fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  /* your code here (hint: useEffect) */
+  useEffect(() => {
+    async function getUserDetails(userId: number) {
+      try {
+        const response = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`
+        );
+        if (!response.ok) setError(response.status);
```

Comment:
This is OK, but it doesn't do your normal exception handling. Also, in future exercises you'll have this code outside of a React component, so you won't be able to `setError`. It's best to throw an exception here.